File: app/src/main/java/com/labessence/biotech/finaldilution/component/ConcentrationType.java
package com.labessence.biotech.finaldilution.component;

public enum ConcentrationType {
    PERCENTAGE(0),
    MOLAR(1),
    MILIMOLAR(2),
    MILIGRAM_PER_MILLILITER(3);

    private final int value;

    ConcentrationType(int value) {
        this.value = value;
    }

    public int getValue() {
        return value;
    }

    @Override
    public String toString() {
        switch (this.value) {
            case 0:
                return "%";
            case 1:
                return "M";
            case 2:
                return "mM";
            case 3:
                return "mg/ml";
            default:
                return "";
        }
    }
}

File: app/src/main/java/com/labessence/biotech/finaldilution/component/concentration/ConcentrationType.kt
package com.labessence.biotech.finaldilution.component.concentration

enum class ConcentrationType(val value: Int) {
    PERCENTAGE(0),
    MOLAR(1),
    MILIMOLAR(2),
    MILIGRAM_PER_MILLILITER(3);

    fun hint(): String {
        return when (this.value) {
            0 -> "%"
            1 -> "M/l"
            2 -> "mM/l"
            3 -> "mg/ml"
            else -> ""
        }
    }

    override fun toString(): String {
        return when (this.value) {
            0 -> "%"
            1 -> "M"
            2 -> "mM"
            3 -> "mg/ml"
            else -> ""
        }
    }

    companion object {
        private val map = ConcentrationType.values().associateBy(ConcentrationType::value)
        fun fromInt(type: Int) = map[type]
    }

}
